src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,121 @@
+    private final StorageDao storageDao = new StorageDaoImpl();
+
+    @Test
+    public void registerNullNotOk() {
+        assertThrows(RegistrationException.class, () -> registrationService.register(null));
+    }
+
+    @Test
+    public void registerNullLoginNotOk() {
+        User user = getValidUser();
+        user.setLogin(null);
+        assertThrows(RegistrationException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    public void registerNullPasswordNotOk() {
+        User user = getValidUser();
+        user.setPassword(null);
+        assertThrows(RegistrationException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    public void registerNullAgeNotOk() {
+        User user = getValidUser();
+        user.setAge(null);
+        assertThrows(RegistrationException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    public void registerAddLoginNotOk() {
+        User user = getValidUser();
+        StringBuilder login = new StringBuilder();
+        user.setLogin(login.toString());
+
+        assertThrows(RegistrationException.class, () -> registrationService.register(user));
+        for (int i = 1; i < MIN_LOGIN_LENGTH; i++) {
+            user.setLogin(login.append(i).toString());
+            assertThrows(RegistrationException.class, () -> registrationService.register(user));
+        }
+    }
+
+    @Test
+    public void registerAlredyRegisteredUserNotOk() {
+        User user = getValidUser();
+        storageDao.add(user);
+
+        assertThrows(RegistrationException.class, () -> registrationService.register(user));
+
+    }
+
+    @Test
+    public void registerAddPasswordNotOk() {
+        User user = getValidUser();
+        StringBuilder password = new StringBuilder();
+        user.setPassword(password.toString());
+
+        assertThrows(RegistrationException.class, () -> registrationService.register(user));
+        for (int i = 1; i < MIN_PASSWORD_LENGTH; i++) {
+            user.setPassword(password.append(i).toString());
+            assertThrows(RegistrationException.class, () -> registrationService.register(user));
+        }
+    }
+
+    @Test
+    public void registerAddAgeNotOk() {
+        User user = getValidUser();
+
+        for (int i = 0; i < MIN_AGE; i++) {
+            user.setAge(i);
+            assertThrows(RegistrationException.class, () -> registrationService.register(user));
+        }
+    }
+
+    @Test
+    public void registerAddOk() {
+        User user = getValidUser();
+        assertEquals(registrationService.register(user), user);
+
+        for (int i = 0; i <= 100; i++) {
+            User newUser = new User();
+            newUser.setLogin(user.getLogin() + i);
+            newUser.setPassword(user.getPassword() + i);
+            newUser.setAge(user.getAge() + i);
+            assertEquals(registrationService.register(newUser), newUser);
+        }
```

Comment:
The test `registerAddOk` does not follow the naming convention. It should be `register_validUser_ok`. Also, this test seems to be relying on the other tests to fill the data, which violates the principle that test cases should be independent.